src/api/app/controllers/webui/packages/bs_requests_controller.rb

```diff
@@ -27,11 +27,12 @@ def index
       private
 
       def filter_by_direction(direction)
-        target = BsRequest.with_actions.where(bs_request_actions: { target_project: @project.name, target_package: @package.name })
-        source = BsRequest.with_actions.where(bs_request_actions: { source_project: @project.name, source_package: @package.name })
+        target = BsRequest.with_actions_and_reviews.where(bs_request_actions: { target_project: @project.name, target_package: @package.name })
+        source = BsRequest.with_actions_and_reviews.where(bs_request_actions: { source_project: @project.name, source_package: @package.name })
+        reviews = BsRequest.with_actions_and_reviews.where(reviews: { package: @package })
         case direction
         when 'all'
-          target.or(source)
+          target.or(source).or(reviews)
```

Comment:
```suggestion
          reviews = BsRequest.with_actions_and_reviews.where(reviews: { package: @package })
          target.or(source).or(reviews)
```

What about moving the retrieval of `reviews` to the 'all' section? This way we don't perform the query of the  reviews unnecessarily for the default case 'incoming'.

Reply:
it's doing nothing until you execute the query... but yeah I moved it down.